### ISSUE TITLE
KBMRobot not working

### ISSUE THREAD

**Issue by EarWaxCandy** [Firebottle/Firebot#309]
#### Describe your issue or feature request (details are helpful!):
In v4.0.0-beta.5 i used KBMRobot to control American Truck Simulator. I can't get any game controls keypresses to work now with version v4.0.1

-----------------------------------------------------------------------
#### If you're reporting an issue, *please* fill out & do the following:

- What version of Firebot are you running?
V4.0.1

- What are the steps to reproduce the issue?
1) added game control button
2) changed to KBMRobot 
3) connected and clicked button. nothing happens. also notice that logs shows the first click, but no other clicks afterwards.

- (IMPORTANT) Go to Firebot's root folder (**Settings tab -> Open Root Folder**) then zip the following two files and attach that zip to this issue:

1) In **user-settings > controls**, find the .json file with the name of the board you are having issues with.
{
    "versionid": 48371,
    "mixer": [
        {
            "sceneID": "default",
            "controls": [
                {
                    "kind": "button",
                    "controlID": "E-Brake",
                    "position": [
                        {
                            "size": "large",
                            "width": 12,
                            "height": 4,
                            "x": 0,
                            "y": 1
                        }
                    ],
                    "text": "E-Brake",
                    "cost": 100
                },
                {
                    "kind": "button",
                    "controlID": "Horn",
                    "position": [
                        {
                            "size": "large",
                            "width": 12,
                            "height": 4,
                            "x": 13,
                            "y": 1
                        }
                    ],
                    "text": "Honk Horn"
                },
                {
                    "kind": "button",
                    "controlID": "Change Cam",
                    "position": [
                        {
                            "size": "large",
                            "width": 12,
                            "height": 4,
                            "x": 26,
                            "y": 1
                        }
                    ],
                    "text": "Change View",
                    "cost": 100
                },
                {
                    "kind": "button",
                    "controlID": "Flash Brights",
                    "position": [
                        {
                            "size": "large",
                            "width": 12,
                            "height": 4,
                            "x": 39,
                            "y": 1
                        }
                    ],
                    "text": "Flash Brights"
                },
                {
                    "kind": "button",
                    "controlID": "Knock Knock",
                    "position": [
                        {
                            "size": "large",
                            "width": 12,
                            "height": 4,
                            "x": 52,
                            "y": 1
                        }
                    ],
                    "text": "Knock Knock",
                    "cost": 100
                },
                {
                    "kind": "button",
                    "controlID": "SkipSong",
                    "position": [
                        {
                            "size": "large",
                            "width": 12,
                            "height": 4,
                            "x": 65,
                            "y": 1
                        }
                    ],
                    "text": "Skip Song",
                    "cost": 5000
                },
                {
                    "kind": "button",
                    "controlID": "CandyParty",
                    "position": [
                        {
                            "size": "large",
                            "width": 12,
                            "height": 7,
                            "x": 65,
                            "y": 6
                        }
                    ],
                    "text": "Candy Party",
                    "cost": 3000
                },
                {
                    "kind": "button",
                    "controlID": "Music Controls",
                    "position": [
                        {
                            "size": "large",
                            "x": 0,
                            "y": 16,
                            "width": 12,
                            "height": 4
                        }
                    ],
                    "text": "Music Controls"
                },
                {
                    "kind": "button",
                    "controlID": "Inspiration",
                    "position": [
                        {
                            "size": "large",
                            "x": 0,
                            "y": 6,
                            "width": 12,
                            "height": 7
                        }
                    ],
                    "text": "Inspiration",
                    "cost": 10
                }
            ]
        },
        {
            "sceneID": "Music Controls",
            "controls": [
                {
                    "kind": "button",
                    "controlID": "Vol 0",
                    "position": [
                        {
                            "size": "large",
                            "width": 6,
                            "height": 4,
                            "x": 2,
                            "y": 5
                        }
                    ],
                    "text": "||"
                },
                {
                    "kind": "button",
                    "controlID": "Vol 10",
                    "position": [
                        {
                            "size": "large",
                            "x": 9,
                            "y": 0,
                            "width": 6,
                            "height": 4
                        }
                    ],
                    "text": "1"
                },
                {
                    "kind": "button",
                    "controlID": "Vol 20",
                    "position": [
                        {
                            "size": "large",
                            "x": 16,
                            "y": 0,
                            "width": 6,
                            "height": 4
                        }
                    ],
                    "text": "2"
                },
                {
                    "kind": "button",
                    "controlID": "Vol 30",
                    "position": [
                        {
                            "size": "large",
                            "x": 23,
                            "y": 0,
                            "width": 6,
                            "height": 4
                        }
                    ],
                    "text": "3"
                },
                {
                    "kind": "button",
                    "controlID": "Vol 40",
                    "position": [
                        {
                            "size": "large",
                            "x": 30,
                            "y": 0,
                            "width": 6,
                            "height": 4
                        }
                    ],
                    "text": "4"
                },
                {
                    "kind": "button",
                    "controlID": "Vol 50",
                    "position": [
                        {
                            "size": "large",
                            "x": 37,
                            "y": 0,
                            "width": 6,
                            "height": 4
                        }
                    ],
                    "text": "5"
                },
                {
                    "kind": "button",
                    "controlID": "Vol 60",
                    "position": [
                        {
                            "size": "large",
                            "width": 6,
                            "height": 4,
                            "x": 9,
                            "y": 5
                        }
                    ],
                    "text": "6"
                },
                {
                    "kind": "button",
                    "controlID": "Vol 70",
                    "position": [
                        {
                            "size": "large",
                            "width": 6,
                            "height": 4,
                            "x": 16,
                            "y": 5
                        }
                    ],
                    "text": "7"
                },
                {
                    "kind": "button",
                    "controlID": "Vol 80",
                    "position": [
                        {
                            "size": "large",
                            "width": 6,
                            "height": 4,
                            "x": 23,
                            "y": 5
                        }
                    ],
                    "text": "8"
                },
                {
                    "kind": "button",
                    "controlID": "Vol 90",
                    "position": [
                        {
                            "size": "large",
                            "width": 6,
                            "height": 4,
                            "x": 30,
                            "y": 5
                        }
                    ],
                    "text": "9"
                },
                {
                    "kind": "button",
                    "controlID": "Vol 100",
                    "position": [
                        {
                            "size": "large",
                            "width": 6,
                            "height": 4,
                            "x": 37,
                            "y": 5
                        }
                    ],
                    "text": "10"
                },
                {
                    "kind": "button",
                    "controlID": "play",
                    "position": [
                        {
                            "size": "large",
                            "width": 6,
                            "height": 4,
                            "x": 2,
                            "y": 0
                        }
                    ],
                    "text": "►"
                },
                {
                    "kind": "button",
                    "controlID": "MainBoard",
                    "position": [
                        {
                            "size": "large",
                            "x": 0,
                            "y": 16,
                            "width": 12,
                            "height": 4
                        }
                    ],
                    "text": "Main Board"
                }
            ]
        }
    ],
    "firebot": {
        "controls": {
            "Change Cam": {
                "controlId": "Change Cam",
                "text": "Change View",
                "cooldown": "",
                "effects": {
                    "1": {
                        "type": "Game Control",
                        "press": "9",
                        "opposite": "",
                        "holding": "No"
                    }
                },
                "scene": "default",
                "cost": 100,
                "kind": "button"
            },
            "Horn": {
                "controlId": "Horn",
                "text": "Honk Horn",
                "cooldown": "",
                "effects": {
                    "1": {
                        "type": "Game Control",
                        "press": "n",
                        "holding": "Yes"
                    }
                },
                "scene": "default",
                "kind": "button"
            },
            "Flash Brights": {
                "controlId": "Flash Brights",
                "text": "Flash Brights",
                "cooldown": "",
                "effects": {
                    "1": {
                        "type": "Game Control",
                        "press": "j",
                        "holding": "No"
                    }
                },
                "scene": "default",
                "kind": "button"
            },
            "E-Brake": {
                "controlId": "E-Brake",
                "text": "E-Brake",
                "cost": 100,
                "cooldown": "60",
                "effects": {
                    "1": {
                        "type": "Game Control",
                        "press": "space"
                    }
                },
                "scene": "default",
                "kind": "button"
            },
            "Knock Knock": {
                "controlId": "Knock Knock",
                "text": "Knock Knock",
                "cost": 100,
                "cooldown": "120",
                "effects": {
                    "1": {
                        "type": "Play Sound",
                        "file": "C:\\Users\\Jeff\\Documents\\ScorpBot\\SFX\\knock knock.mp3",
                        "volume": "10"
                    }
                },
                "scene": "default",
                "kind": "button"
            },
            "SkipSong": {
                "controlId": "SkipSong",
                "text": "Skip Song",
                "cost": 5000,
                "cooldown": "300",
                "scene": "default",
                "kind": "button"
            },
            "CandyParty": {
                "controlId": "CandyParty",
                "text": "Candy Party",
                "cost": 3000,
                "cooldown": "30",
                "effects": {
                    "1": {
                        "type": "Chat",
                        "chatter": "Streamer",
                        "message": "Candy for everyone! Thank $(user)",
                        "whisper": ""
                    },
                    "2": {
                        "type": "Celebration",
                        "celebration": "Fireworks",
                        "length": "10"
                    }
                },
                "scene": "default",
                "kind": "button"
            },
            "Vol 0": {
                "controlId": "Vol 0",
                "text": "||",
                "cooldown": "",
                "effects": {
                    "1": {
                        "type": "Chat",
                        "chatter": "Streamer",
                        "message": "!pause",
                        "whisper": "earwaxbot"
                    }
                },
                "scene": "Music Controls",
                "kind": "button"
            },
            "Vol 10": {
                "controlId": "Vol 10",
                "text": "1",
                "cooldown": "",
                "effects": {
                    "1": {
                        "type": "Chat",
                        "chatter": "Streamer",
                        "message": "!setvolume 10",
                        "whisper": "earwaxbot"
                    }
                },
                "scene": "Music Controls",
                "kind": "button"
            },
            "Vol 20": {
                "controlId": "Vol 20",
                "text": "2",
                "cooldown": "",
                "effects": {
                    "1": {
                        "type": "Chat",
                        "chatter": "Streamer",
                        "message": "!setvolume 20",
                        "whisper": "earwaxbot"
                    }
                },
                "scene": "Music Controls",
                "kind": "button"
            },
            "Vol 30": {
                "controlId": "Vol 30",
                "text": "3",
                "cooldown": "",
                "effects": {
                    "1": {
                        "type": "Chat",
                        "chatter": "Streamer",
                        "message": "!setvolume 30",
                        "whisper": "earwaxbot"
                    }
                },
                "scene": "Music Controls",
                "kind": "button"
            },
            "Vol 40": {
                "controlId": "Vol 40",
                "text": "4",
                "cooldown": "",
                "effects": {
                    "1": {
                        "type": "Chat",
                        "chatter": "Streamer",
                        "message": "!setvolume 40",
                        "whisper": "earwaxbot"
                    }
                },
                "scene": "Music Controls",
                "kind": "button"
            },
            "Vol 50": {
                "controlId": "Vol 50",
                "text": "5",
                "cooldown": "",
                "effects": {
                    "1": {
                        "type": "Chat",
                        "chatter": "Streamer",
                        "message": "!setvolume 50",
                        "whisper": "earwaxbot"
                    }
                },
                "scene": "Music Controls",
                "kind": "button"
            },
            "Vol 60": {
                "controlId": "Vol 60",
                "text": "6",
                "cooldown": "",
                "effects": {
                    "1": {
                        "type": "Chat",
                        "chatter": "Streamer",
                        "message": "!setvolume 60",
                        "whisper": "earwaxbot"
                    }
                },
                "scene": "Music Controls",
                "kind": "button"
            },
            "Vol 70": {
                "controlId": "Vol 70",
                "text": "7",
                "cooldown": "",
                "effects": {
                    "1": {
                        "type": "Chat",
                        "chatter": "Streamer",
                        "message": "!setvolume 70",
                        "whisper": "earwaxbot"
                    }
                },
                "scene": "Music Controls",
                "kind": "button"
            },
            "Vol 80": {
                "controlId": "Vol 80",
                "text": "8",
                "cooldown": "",
                "effects": {
                    "1": {
                        "type": "Chat",
                        "chatter": "Streamer",
                        "message": "!setvolume 80",
                        "whisper": "earwaxbot"
                    }
                },
                "scene": "Music Controls",
                "kind": "button"
            },
            "Vol 90": {
                "controlId": "Vol 90",
                "text": "9",
                "cooldown": "",
                "effects": {
                    "1": {
                        "type": "Chat",
                        "chatter": "Streamer",
                        "message": "!setvolume 90",
                        "whisper": "earwaxbot"
                    }
                },
                "scene": "Music Controls",
                "kind": "button"
            },
            "Vol 100": {
                "controlId": "Vol 100",
                "text": "10",
                "cooldown": "",
                "effects": {
                    "1": {
                        "type": "Chat",
                        "chatter": "Streamer",
                        "message": "!setvolume 100",
                        "whisper": "earwaxbot"
                    }
                },
                "scene": "Music Controls",
                "kind": "button"
            },
            "Music Controls": {
                "controlId": "Music Controls",
                "text": "Music Controls",
                "cooldown": "5",
                "effects": {
                    "1": {
                        "type": "Change Scene",
                        "scene": "Music Controls",
                        "groups": [
                            "Moderators"
                        ],
                        "reset": false
                    }
                },
                "scene": "default",
                "kind": "button"
            },
            "play": {
                "controlId": "play",
                "text": "►",
                "cooldown": "",
                "effects": {
                    "1": {
                        "type": "Chat",
                        "chatter": "Streamer",
                        "message": "!play",
                        "whisper": "earwaxbot"
                    }
                },
                "scene": "Music Controls",
                "kind": "button"
            },
            "MainBoard": {
                "controlId": "MainBoard",
                "text": "Main Board",
                "cooldown": "5",
                "effects": {
                    "1": {
                        "type": "Change Scene",
                        "scene": "default",
                        "groups": [
                            "default",
                            "Moderators"
                        ],
                        "reset": false
                    }
                },
                "scene": "Music Controls",
                "kind": "button"
            },
            "Inspiration": {
                "controlId": "Inspiration",
                "scene": "default",
                "text": "Inspiration",
                "cost": 10,
                "kind": "button",
                "effects": {
                    "1": {
                        "type": "HTML",
                        "html": "<div id ='bod' class=\"img1\"></div>\n\n<script type=\"text/javascript\" language=\"javascript\">\nvar xhr = new XMLHttpRequest();\nxhr.open(\"GET\", \"http://inspirobot.me/api?generate=true\", false);\nxhr.send();\nconsole.log(xhr.status);\nconsole.log(xhr.statusText);\n\nvar image = document.createElement(\"img\");\nvar imageParent = document.getElementById(\"bod\");\nimage.id = \"id\";\nimage.className = \"class\";\nimage.src = xhr.responseText;\nimage.width=800;\nimageParent.appendChild(image);\n\n</script>",
                        "length": "10",
                        "removal": "img1"
                    }
                },
                "cooldown": "60"
            }
        },
        "scenes": {
            "default": {
                "sceneName": "default",
                "default": []
            },
            "Music Controls": {
                "sceneName": "Music Controls",
                "default": [
                    "Moderators"
                ]
            }
        }
    }
}

2) In **user-settings > logs**, find the latest log file.
No log file present

```
NOTE: A lot of issues can be solved by,
- Updating to the latest version
- Trying the troubleshooting steps on the wiki
```

**Comment by Firebottle** [comment]
We'll look into it. 4.0.0 full release changed some of the locations of the KBM robot java file. Likely it is in the wrong place and KBMJava can't find it.

For now you'll need to switch to robotjs.

**Comment by EarWaxCandy** [comment]
robotsjs doesn't seem to work with truck sim :(

**Comment by Firebottle** [comment]
Fix for this in pull request #315

**Comment by Firebottle** [comment]
Released 4.0.2. Just open up a new bug if you continue to have issues.
